src/System.Management.Automation/engine/CommandParameter.cs

```diff
@@ -15,12 +15,14 @@ internal sealed class CommandParameterInternal
     {
         private class Parameter
         {
+            internal Ast ast;
             internal IScriptExtent extent;
```

Comment:
`ast` makes storing the `extent` no longer necessary, right?

Reply:
I'm still not happy with storing 2 references a typical script has a ton of these instances.

What if we use something like:
```c#
    private object context;  // Instead of storing Extent and Ast
    // ...
    internal Ast ParameterAst { return _parameter?.context as Ast; }
    internal IScriptExtent ParameterExtent {
        return _parameter?.context as IScriptExtent ?? (_parameter?.context as Ast)?.Extent;
    }
    // etc.
```